client/question/services/FeedbackGeneratorService.js

```diff
@@ -53,6 +69,23 @@ tie.factory('FeedbackGeneratorService', [
     var consecutiveSameRuntimeErrorCounter = 0;
 
     /**
+     * Object to hold index of correctness feedback given (values), per
+     * test suite (keys). Used when avoiding presenting the same feedback text
+     * consecutively.
```

Comment:
If the only thing we care about here is not showing the same feedback consecutively, is there a reason to not just use a string instead? If a string would suffice, we don't even need to store this here -- we could just extract the previous feedback given from the TranscriptService.

An alternative (if we really want to store the index) is to store two vars: the previous type and the previous index. Only if both match, do we regenerate.

Reply:
The intention is to not show the same feedback twice in a row, even if the user got another piece of feedback in between. I.e., not only is the following avoided:
1. feedback 1 of type a
2. feedback 1 of type a
The following is also avoided:
1. feedback 1 of type a
2. feedback 1 of type b
3. feedback 1 of type a